crates/core/src/table/fs_view.rs

```diff
@@ -70,7 +73,31 @@ impl FileSystemView {
         if partition_paths.is_empty() {
             partition_paths.push("".to_string())
         }
-        Ok(partition_paths)
+        if partition_filters.is_empty() {
+            return Ok(partition_paths);
+        }
+        let field_and_data_type: HashMap<_, _> = partition_schema
+            .fields()
+            .iter()
+            .map(|field| (field.name().to_string(), field.data_type().clone()))
+            .collect();
+
+        Ok(partition_paths
```

Comment:
all partition paths were listed and then filtered - a more optimized approach is to filter out partitions during the listing process, will reduce a lot of scanning for multi-level partitions. we can optimize this in a separate PR

Reply:
good idea, will create a separate PR improve it.